### DISCUSSION TITLE
How to run deno on own host

### DISCUSSION THREAD

**Discussion by andriiradkevych** [freshframework/fresh#580] [General]
As I see from doc https://fresh.deno.dev/docs/getting-started/deploy-to-production , you propose to deploy deno by `Deno Deploy` , but what if I want to set deno on my own host , by starting my app in production mod , how can i make it ?

```
deno run -A main.ts
```

But I'm not sure is it what I need to make it work in best way .. Also I see there is no build step , is it build code so fast that you say there is no build , or there is build on the fly ?

**Comment by excessivecoding** [comment]
I have not playing with self-hosting myself, but you could maybe run something like Supervisor, pm2 or something similar to execute your command and ensure restart on crash.

Personally, I tend to go more for Docker, but the setup might be harder if you are not familiar with it.

From my understanding, there is no build step as running `deno run -A main.ts` on your server will work without you doing any process beforehand, the only thing generated is the `fresh.gen.ts` which is just mapping out resources for your application to run but that's about it.
